Spigot-API-Patches/0283-Add-ElderGuardianAppearanceEvent.patch

```diff
@@ -0,0 +1,76 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Owen1212055 <23108066+Owen1212055@users.noreply.github.com>
+Date: Fri, 19 Mar 2021 23:39:21 -0400
+Subject: [PATCH] Add ElderGuardianAppearanceEvent
+
+
+diff --git a/src/main/java/io/papermc/paper/event/entity/ElderGuardianAppearanceEvent.java b/src/main/java/io/papermc/paper/event/entity/ElderGuardianAppearanceEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..f7d41c4db09d95a5f8f5b64b9adf1951c894f4d9
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/entity/ElderGuardianAppearanceEvent.java
+@@ -0,0 +1,64 @@
++package io.papermc.paper.event.entity;
++
++import org.bukkit.entity.ElderGuardian;
++import org.bukkit.entity.Entity;
++import org.bukkit.entity.Player;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.entity.EntityEvent;
++import org.jetbrains.annotations.NotNull;
++
++import java.util.Collections;
++import java.util.List;
++
++/**
++ * Is called when an {@link org.bukkit.entity.ElderGuardian} appears in front of a {@link org.bukkit.entity.Player}.
++ */
++public class ElderGuardianAppearanceEvent extends EntityEvent implements Cancellable {
++
++    private static final HandlerList handlers = new HandlerList();
++
++    private boolean cancelled;
++    private final List<Player> affectedPlayers;
++
++    public ElderGuardianAppearanceEvent(@NotNull Entity what, @NotNull List<Player> affectedPlayers) {
++        super(what);
++        this.affectedPlayers = affectedPlayers;
++    }
++
++    /**
++     * Get a list of players affected by the guardian appearance.
++     *
++     * @return Players affected by the appearance
++     */
++    @NotNull
++    public List<Player> getAffectedPlayers() {
++        return affectedPlayers;
```

Comment:
The mutability of this returned value should be documented. It seems to be mutable at the moment, it should be mentioned that mutating it is a completely valid thing to do and that it actually causes stuff to change.